### ISSUE TITLE
UnusedPrivateProperty false negative with properties on constructor

### ISSUE THREAD

**Issue by BraisGabin** [detekt/detekt#9467]
## Expected Behavior
`UnusedPrivateProperty` points that `private val` is unused.

```kotlin
class Foo(private val text: String) {
    val count2: Int = text.length
}
```

## Observed Behavior
It doesn't

## How to reproduce it
With this unit test
```kotlin
@Test
fun `should report when the parameter is used but no the property`() {
    val code = """
        class Foo(private val text: String) {
            val count2: Int = text.length
        }
    """.trimIndent()
    val findings = subject.lintWithContext(env, code)
    assertThat(findings).hasSize(1)
}
```

## Your Environment
* Version of detekt used: main
